pkg/hnsw/hnsw.go

```diff
@@ -28,17 +28,18 @@ type Hnsw struct {
 	levelMultiplier float64
 }
 
-func NewHNSW(m int, efc int, entryPoint *Node) *Hnsw {
+func NewHNSW(m int, efc int, entryVector Vector) *Hnsw {
```

Comment:
Do we check the dimension of the vector? We might want to add a dim parameter to the HNSW to ensure that all the vectors are the correct dimension to avoid a footgun.

Reply:
That was originally included but I took it out as I was verifying the correctness. 

Will include!